pulpcore/cli/core/context.py

```diff
@@ -15,20 +15,33 @@ class PulpArtifactContext(PulpEntityContext):
     LIST_ID = "artifacts_list"
     READ_ID = "artifacts_read"
 
-    def upload(self, file: IO[bytes], chunk_size: int = 1000000) -> Any:
+    def upload(self, file: IO[bytes], chunk_size: int = 1000000, check_exists: bool = True) -> Any:
         upload_ctx = PulpUploadContext(self.pulp_ctx)
         start = 0
         size = os.path.getsize(file.name)
         sha256 = hashlib.sha256()
-        upload_href = upload_ctx.create(body={"size": size})["pulp_href"]
+
+        if check_exists:
+            while True:
```

Comment:
If only we had Python 3.8 and I could use `while chunk := file.read(chunk_size):` here.

Reply:
I know it's not exactly the same:

https://github.com/theforeman/foreman-ansible-modules/blob/develop/plugins/modules/content_upload.py#L176